### DISCUSSION TITLE
no sound from MIDI controller

### DISCUSSION THREAD

**Discussion by RobyGitH** [GeekFunkLabs/fluidpatcher#75] [Q&A]
hi all, I've got a piano MIDI controller connected to the squishbox (raspberry pi).
1) MIDI controller is recognised and MIDI messages are received as confirmed with aseqdump and on the squishbox LCD ("MIDI monitor")
2) DAC card is recognized and audio works well when played a mp3 file
however, when MIDI controller keys are pressed, there is no sound

aplay -l

**** List of PLAYBACK Hardware Devices ****
card 0: sndrpihifiberry [snd_rpi_hifiberry_dac], device 0: HifiBerry DAC HiFi pcm5102a-hifi-0 [HifiBerry DAC HiFi pcm5102a-hifi-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 2: vc4hdmi [vc4-hdmi], device 0: MAI PCM i2s-hifi-0 [MAI PCM i2s-hifi-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0

aconnect -i

client 0: 'System' [type=kernel]
    0 'Timer           '
    1 'Announce        '
client 14: 'Midi Through' [type=kernel]
    0 'Midi Through Port-0'
client 20: 'Oxygen 49' [type=kernel,card=1]
    0 'Oxygen 49 Oxygen 49Oxygen 49'

I have done updates several times, reinstalled the system through
curl -L git.io/squishbox | bash

but still no sounds when the piano keys are hit
Any ideas? I must have spent 10 hours on this problem and do not know what could be wrong...

any help would be hugely appreciated!
Roby

**Comment by RobyGitH** [comment]
hi all, after checking all setting files, I found the issue
I amended the following line 
audio.alsa.device: hw:sndrpihifiberry
in the file 'squishboxconf.yaml'
not sure why it was picking up vc4hdmi, as I specified sndrpihifiberry when I did the installation

Another question. I noticed that some sounds such as piano FM or others work well, but many struggle with my Pi Zero. Do I need to use a Pi 3B+ or can i get away with my Zero by changing some of the settings?

thanks!
Roberto

thanks anyway.

**Comment by RobyGitH** [comment]
I tried a Pi 3B+ and it works much better. I guess the Pi Zero is good only to play 3-4 simple MIDI sounds at a time (eg piano). 
thanks

**Comment by albedozero** [comment]
Thanks for the information - I'll check for bugs in the install script that might have caused the failed selection.

When you say some sounds don't work well on the Pi Zero, can you be more specific? Is there too much latency, does the sound crackle, or does it just cut out completely? The latter is evidence the CPU is being overloaded, which can be more of a problem on the Pi Zero. The Zero 2 is supposedly close to a Pi 3B+ in terms of processing power, but in practice this doesn't seem to be the case to me. A way to deal with CPU overruns is to lower the `synth.polyphony` setting in _squishboxconf.yaml_ - this tells FluidSynth to drop older notes before the CPU gets overloaded.

**Reply by RobyGitH** [reply]
hi Bill, 
thanks for getting back to me. With the Pi Zero, if I press 4+ keys on the piano with a simple soundfont such as PIANO or if I press just a couple of keys with a more complex soundfont, the sound cuts out completely for 1-2 seconds. It looks like the CPU, as you say, is overloaded, and I am not surprised of this considering the CPU power of the Pi Zero. I use the Pi Zero 2 for other tasks and I do not think that it is close to the Pi 3B+ which I also use, so I tend to agree with you on this. I will try your suggestion with the synth.polyphony setting, but I am very happy with the way the Pi 3B+ works. 

I added a LED to clearly show the squishbox is ON by connecting it to GPIO 14 (+ 10k resistor). I just used the following code:
sudo nano /boot/config.txt
enable_uart=1

Great software, thanks very much for your contribution and your support!
